share/p2p/shrexsub/pubsub.go

```diff
@@ -80,29 +87,52 @@ func (s *PubSub) Stop(context.Context) error {
 	return s.topic.Close()
 }
 
-// AddValidator registers given Validator for EDS notifications (DataHash).
+// AddValidator registers given ValidatorFn for EDS notifications.
 // Any amount of Validators can be registered.
-func (s *PubSub) AddValidator(validate Validator) error {
-	return s.pubSub.RegisterTopicValidator(s.pubsubTopic,
-		func(ctx context.Context, p peer.ID, msg *pubsub.Message) pubsub.ValidationResult {
-			return validate(ctx, p, msg.Data)
-		})
+func (s *PubSub) AddValidator(v ValidatorFn) error {
+	return s.pubSub.RegisterTopicValidator(s.pubsubTopic, v.validate)
+}
+
+func (v ValidatorFn) validate(ctx context.Context, p peer.ID, msg *pubsub.Message) pubsub.ValidationResult {
```

Comment:
I don't understand this - why can't we have this as a standalone function? why is it a method on `ValidatorFn` type?

Reply:
Can we unify all this logic into one validation func?

Reply:
This function handles wire msg decoding and validation and then passes `Notification` to ValidatorFn (line 111) to get validation result. it could be a free function, but still would need ValidatorFn as an input. We can't make it one validation, because Validator is passed from external pkg.  Maybe you have ideas for better naming?